src/Compiler/Checking/NicePrint.fs

```diff
@@ -2615,6 +2615,8 @@ let stringOfFSAttrib denv x = x |> PrintTypes.layoutAttrib denv |> squareAngleL
 
 let stringOfILAttrib denv x = x |> PrintTypes.layoutILAttrib denv |> squareAngleL |> showL
 
+let fqnOfEntityRef (tcref: EntityRef) = $"""{String.Join(".", tcref.CompilationPath.DemangledPath)}.{tcref.LogicalName}"""
```

Comment:
I would reuse the existing function for this:

```fsharp
let fqnOfEntityRef g (tcref: EntityRef) =
    layoutTyconRefImpl false (DisplayEnv.Empty g) tcref
    |> showL
```

you could argue to also pass `nenv.DisplayEnv` from the call site.

Reply:
I went with passing `TcGlobals` to make sure we get the FQN in the output. Passing `nenv.DisplayEnv` would shorten the names.

Reply:
Probably shove it behind a langversion flag, less than preview or 8 make it an info, above, either a warning or an error.

Reply:
@KevinRansom Sorry, I'm a bit confused here, as preview is defined as
`static let previewVersion = 9999m // Language version when preview specified`
all future releases will be "less than preview or 8" and we'll never move to the warning/error stage.
How about:
```fsharp
let langV = LanguageVersion(g.langVersion.VersionText)
    if langV.SpecifiedVersion > 8.0m then
        warning(...)
    else
        informationalWarning(...)
```

Reply:
I think Kevin meant to guard it via our default SupportsFeature mechanism, and put this feature to preview version (which will become 8 eventually).